src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -686,5 +701,71 @@ internal LocalDeclarations TryGetLocalDeclarations(FileContentManager file, Posi
             var declarations = implementation?.LocalDeclarationsAt(pos.Subtract(specPos), includeDeclaredAtPosition);
             return this.PositionedDeclarations(parentCallable, callablePos, specPos, declarations); 
         }
+
+        /// <summary>
+        /// Returns a short tag name for the given source file. The tag is not unique. If multiple source files are
+        /// being tagged, make sure there are no name conflicts.
+        /// </summary>
+        /// <param name="source">The source file to get the tag for.</param>
+        /// <returns>A short tag name for the given source file.</returns>
+        internal static string GetTagForSource(string source) =>
```

Comment:
I'd make this a static method in RenameReferences, and the same for EmbedTag.

Reply:
RenameReferences is meant to be a general transformation that doesn't care what kind of renaming it is doing, so I put the name-mangling logic in the Compilation class, since that is where we do care that the mangling is done in a certain way (based on source file).

Reply:
Actually, I should probably unify this with GenerateUniqueName and StripUniqueName in the UniqueVariableNames transformation.

Reply:
I updated it to use the same name format as UniqueVariableNames, could you take another look?

Reply:
Done.